ipa-core/src/protocol/ipa_prf/aggregation/mod.rs

```diff
@@ -372,17 +372,17 @@ pub mod tests {
                 Ok(input_row(1, &[0, 0, 0, 0, 1, 1, 1, 1])),
                 Ok(input_row(1, &[0, 0, 0, 0, 1, 0, 1, 1])),
             ];
-
-            let result = TestWorld::default()
+            let result: BitDecomposed<BA8> = TestWorld::default()
                 .upgraded_semi_honest(inputs.into_iter(), |ctx, inputs| {
                     let num_rows = inputs.len();
                     aggregate_values::<BA8, 8>(ctx, stream::iter(inputs).boxed(), num_rows)
                 })
                 .await
                 .map(Result::unwrap)
-                .reconstruct();
-
-            assert_eq!(result, [0u32, 1, 2, 3, 4, 5, 6, 7].map(BA8::truncate_from));
+                .reconstruct_arr();
+            let expected: BitDecomposed<BA8> = input_row(8, &[0u32, 1, 2, 3, 4, 5, 6, 7])
```

Comment:
Wait, shouldn't we expect a transposed result now

Reply:
yes, so here `input_row` applies transpose (turning into vectorized form) to the expected result.  So the `result` is still in vectorized from when we compare it in the `assert_eq`